File: Strings/1_10/8Solution.py
def minTimeToType(word):
        l = list(word)
        res = len(l)
        w = 'a'
        for i in l:
            k = abs(ord(i) - ord(w))
            if k>13:
                k = 26-k
            res += k
            w = i
        return res
word=input()
print(minTimeToType(word))

File: Strings/1_10/8Solution.java
class Solution {
    public int minTimeToType(String word) {
        int result=word.length();
        char curr='a';
        for(char c:word.toCharArray()){
            int time=c-curr;
            if(c<curr)
                time=curr-c;
            time=Math.min(time,26-time);
            result+=time;
            curr=c;
        }
        return result;
    }
}
